plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/AlertsData.java

```diff
@@ -0,0 +1,93 @@
+package edu.wpi.first.shuffleboard.plugin.base.data;
+
+import edu.wpi.first.shuffleboard.api.data.ComplexData;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import java.util.Map;
+
+public final class AlertsData extends ComplexData<AlertsData> {
+
+  private final String[] errors;
+  private final String[] warnings;
+  private final String[] infos;
+
+  /**
+   * Creates a new AlertsData object.
+   *
+   * @param errors   List of active error alerts
+   * @param warnings List of active warning alerts
+   * @param infos    List of active info alerts
+   */
+  public AlertsData(String[] errors, String[] warnings, String[] infos) {
+    this.errors = errors;
+    this.warnings = warnings;
+    this.infos = infos;
+  }
+
+  /**
+   * Gets the list of error alerts.
+   */
+  public String[] getErrors() {
+    return errors;
+  }
+
+  /**
+   * Gets the list of warning alerts.
+   */
+  public String[] getWarnings() {
+    return warnings;
+  }
+
+  /**
+   * Gets the list of info alerts.
+   */
+  public String[] getInfos() {
+    return infos;
+  }
+
+  /**
+   * Gets a collection of all alerts.
+   */
+  public ObservableList<AlertItem> getCollection() {
+    ObservableList<AlertItem> collection = FXCollections.observableArrayList();
+    for (String text : errors) {
+      collection.add(new AlertItem(AlertType.ERROR, text));
+    }
+    for (String text : warnings) {
+      collection.add(new AlertItem(AlertType.WARNING, text));
+    }
+    for (String text : infos) {
+      collection.add(new AlertItem(AlertType.INFO, text));
+    }
+    if (collection.size() == 0) {
+      collection.add(new AlertItem(AlertType.LOCAL, "(Nothing to report)"));
+    }
```

Comment:
This would be better as a placeholder[1] on the widget instead of being baked into the data object. It would also make the `LOCAL` alert type unnecessary

[1] https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/ListView.html#placeholderProperty--

Reply:
Fixed, with a slight style tweak so that the background color doesn't change when the list disappears